litestar/openapi/config.py

```diff
@@ -94,6 +94,13 @@ class OpenAPIConfig:
     """A set of the enabled documentation sites and schema download endpoints."""
     operation_id_creator: OperationIDCreator = default_operation_id_creator
     """A callable that generates unique operation ids"""
+    path: str = "/schema"
+    """Base path for the OpenAPI documentation endpoints."""
+
+    def __post_init__(self) -> None:
+        if self.openapi_controller is OpenAPIController:
```

Comment:
So what's the point here?

Reply:
It's to make sure we respect the custom OpenAPIController's path and only override the path for the default OpenAPIController, so:
```python
class MyOpenAPIController:
    path = "/customdocs"

app = Litestar([], openapi_config=OpenAPIConfig(title="Litestar API", version="1.0.0"), openapi_controller=MyOpenAPIController)
```
would still serve docs at `/customdocs` instead of `/schema`.

Does it feel hacky?

Reply:
Why not simply check if the controller class has path defined on it? Use getattr etc

Reply:
Is this something we should worry about? I believe a path customization would be overwritten in this case - it is a bit contrived though.

```py
from litestar.openapi import OpenAPIConfig, OpenAPIController

OpenAPIController.path = "/somewhere-else"

app = Litestar(..., openapi_config=OpenAPIConfig(openapi_controller=OpenAPIController))
```
What about making `OpenAPIConfig.path` `str | None = None` and only subclass and overwrite the path if the config path is not `None`? This would mean that a config defined path would always take precedence over a controller defined path.

Reply:
Fair enough, but in this case we need validation

Reply:
For validation, aside from enforcing the path starts with `/`, is there anything else?

Reply:
We have to validate the path is actually set. 

Furthermore, it cannot be `/` only but must be a proper sub path.

Reply:
thanks! I'll get it in sometime this weekend